src/Trains.NET.Rendering/Trains/TrainPainter.cs

```diff
@@ -19,18 +18,18 @@ public class TrainPainter : ITrainPainter
 
     public TrainPalette GetPalette(Train train)
     {
-        if (!_paletteMap.ContainsKey(train.UniqueID))
+        if (!_paletteMap.ContainsKey(train.Seed))
         {
-            _paletteMap.Add(train.UniqueID, GetRandomPalette());
+            _paletteMap.Add(train.Seed, GetRandomPalette(train.GetPRNG()));
```

Comment:
This isn't random any more, right? so aside from renaming the method, do we need a cache at all?

Reply:
Good pickup, renamed the method, it's probably a 50/50 split as to if it needs to be cached, there is some ToString's and Replace when building the palette, so probably still worthwhile keeping the caching in.